### DISCUSSION TITLE
ErrorFolderNotFound: No subfolder with name "***"

### DISCUSSION THREAD

**Discussion by kendrick-brown** [ecederstrand/exchangelib#1157] [Q&A]
I am getting the above Error when attempting to access emails in a subfolder within Public Folders even though the subfolder exists. It has worked with no issues for months until yesterday. Please find at the bottom the code snippet as well as the error message snippet. Any insight on what could be causing that would be greatly appreciated. 

Also going off an issue I came across (that may be related) here, [https://stackoverflow.com/questions/52145707/python-exchangelib-how-to-access-public-folders](url). If I run "account.public_folder_root.absolute" I get '/IPM_SUBTREE'.

Let me know if more details are needed. Thanks in advance!

![image](https://user-images.githubusercontent.com/84810485/211638775-e6a6b2fd-49b1-45fb-b6de-96e95266d8b8.png)

![image](https://user-images.githubusercontent.com/84810485/211639276-22b7c398-18d7-480f-a9d5-627277ac8af4.png)

**Comment by bn-crandom** [comment]
For what it's worth, public folders stopped working for us on 1/7.  Worked fine on 1/6.  I tried different versions of exchangelib and got the same outcome-- the public folder root has no children, even though the M365 tenant has public folders (and exchangelib worked fine with them previously).  I suspect something may have changed on the M365 end.

**Comment by natenjo** [comment]
We can confirm this issue.  It also stopped working in the first days of January 2023.

We first suspected the cause was the depreciation of other authenfication methods, so we updated to oAuth2 ([see Microsoft documentation](https://learn.microsoft.com/en-us/exchange/clients-and-mobile-in-exchange-online/deprecation-of-basic-authentication-exchange-online#when-will-this-change-take-place)).

The account has permissions to the public folder structure and worked at least until end of 2022.

After setting this up, the Error was still the same.

**Sample code:**
```
credentials = OAuth2Credentials(
 client_id='***', 
 client_secret='***', 
 tenant_id='***',
 identity=Identity(primary_smtp_address='***')
)

config = Configuration(
  server='outlook.office365.com',
  credentials=credentials,
)

my_account = Account(
  primary_smtp_address='***', 
  config=config,
  autodiscover=False, 
  access_type=DELEGATE
)

my_account.root.refresh()
my_account.public_folders_root.refresh()
print(my_account.public_folders_root.tree())
```

Result:
```
root@ubuntu:~# python3 test.py
IPM_SUBTREE
```

We are using  the current up to date version of the libary ( v4.9.0)

Any help/hint is appreciated

**Comment by natenjo** [comment]
I created a Issue for this topic, see https://github.com/ecederstrand/exchangelib/issues/1160

**Comment by natenjo** [comment]
For us it is working again, I assume Microsoft "fixed" something..I also updated/closed  the issue I  opened ( https://github.com/ecederstrand/exchangelib/issues/1160 )

**Reply by ecederstrand** [reply]
Thanks for the confirmation! If this happens again, we'll need a dump of the response XML. I have a suspicion that this happens because the error message started appeared in a different part of the XML document than we expected, and thus wasn't caught.

**Reply by kendrick-brown** [reply]
Thank you, it appears to back working on my end as well.
